sdk/rust/oak_runtime_compat/src/asylo/allocator.rs

```diff
@@ -73,3 +76,36 @@ unsafe impl GlobalAlloc for System {
         free(ptr);
     }
 }
+
+#[link(name = "sgx_trts")]
+extern "C" {
+    // SGX-enabled abort function that causes an undefined instruction (`UD2`) to be executed, which
+    // terminates the enclave execution.
+    //
+    // The C type of this function is `extern "C" void abort(void) __attribute__(__noreturn__);`
+    //
+    // See https://github.com/intel/linux-sgx/blob/d166ff0c808e2f78d37eebf1ab614d944437eea3/sdk/trts/linux/trts_pic.S#L565.
+    fn abort() -> !;
+}
```

Comment:
Should this be in its own crate, or at least file? It's not exclusive to the allocator, we may need it for other use cases too.

Reply:
Hmm good question. I'll move it to `asylo/mod.rs`, and if it needs to be exported further then we can add a backend independent function reexporting it in `lib.rs`